packages/admin-ui-extensions/src/components/ResourcePicker/ResourcePicker.ts

```diff
@@ -0,0 +1,145 @@
+import {createRemoteComponent} from '@remote-ui/core';
+import {DestructableAction} from '../types';
+
+type Alignment = 'leading' | 'trailing' | 'center';
+
+interface AvatarProps {
+  kind: 'Avatar';
+  accessibilityLabel: string;
+  name: string;
+  initials: string;
+  source: string;
+}
+
+interface ThumbnailProps {
+  kind: 'Thumbnail';
+  source: string;
+  alt: string;
+}
+
+interface FilterControl {
```

Comment:
This feels slightly opinionated but I guess we could always make this a React.Element or similar later if they need more flexibility.